### PR TITLE
refactor: extract several parsing utilities

### LINKED ISSUE #9: Improve code readability in workflow.py:from_dict
The code checks the type of "uses" in the workflow - We can do it more elegantly.
```
if isinstance(d["on"], str):
            trigger = [d["on"]]
        elif isinstance(d["on"], list):
            trigger = []
            for elem in d["on"]:
                if isinstance(elem, dict):
                    trigger.extend(elem.keys())
                else:
                    trigger.append(elem)
        elif isinstance(d["on"], dict):
```

### PR DESCRIPTION
**Pull request by alex-ilgayev**
Fixes #9

### CODE REVIEW

File: src/workflow_components/parsing_utils.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        } -> ["ubuntu-latest"]
+    """
+    if isinstance(runs_on_obj, str):
+        return [runs_on_obj]
+    elif isinstance(runs_on_obj, list):
+        return runs_on_obj
+    elif isinstance(runs_on_obj, dict):
+        return runs_on_obj["labels"]
+
+    return None
```
Comment (oreenlivnicode) [inline src/workflow_components/parsing_utils.py:59]: Is this a convention?

File: src/workflow_components/workflow.py [hunk 2]
```diff
@@ -170,40 +171,29 @@ def __init__(self, name: Optional[str], path: str):
...
-                    trigger.extend(elem.keys())
+        w.trigger = parse_workflow_trigger(obj_dict["on"])
+
+        # Handling special case of workflow_run
+        # When we meet it, we want to create a special relation from the triggering workflow,
+        # to the triggered one.
+        # There are cases where the triggering workflow wasn't loaded yet.
+        # In that case we creating a stub node for it,
+        # and once we'll meet it, we'll enrich it.
+        if "workflow_run" in w.trigger:
```
Comment (oreenlivnicode) [inline src/workflow_components/workflow.py:182]: Can this be in paring function as well?

File: tests/unit/test_parsing_utils.py
Comment (oreenlivnicode) [inline tests/unit/test_parsing_utils.py:1]: Pro🚀

File: src/workflow_components/workflow.py (hunk 2 above)
Comment (alex-ilgayev) [reply src/workflow_components/workflow.py:182]: I thought deeply about it.
Once we find the workflow run trigger, we need to find the workflow and update the relations
So the parsing function will be either 3 lines (which is useless), or receive many parameters (which will be harder to test).
I prefer to leave it as is

File: src/workflow_components/parsing_utils.py (hunk 1 above)
Comment (alex-ilgayev) [reply src/workflow_components/parsing_utils.py:59]: What specifically?

Comment (oreenlivnicode) [reply src/workflow_components/parsing_utils.py:59]: Why do you explicitly return None? It will happen anyway

**Pull request merged**